specifications/api/swagger.yaml

```
@@ -3717,18 +3729,13 @@ components:
             asset_name: *assetName
             operation: *ApiMintBurnOperation
 
-    ApiVoteAction: &ApiVoteAction
+    ApiDRep: &ApiDRep
+      <<: *anyVoting
       description: |
         Voting action. One can abstain, give no confidence vote
         or vote for a representative by specifying its key hash or script hash.
-        Voting can be done together with delegation action.
-      oneOf:
-        - type: string
-          enum:
-            - abstain
-            - no_confidence
-        - type: string
-          <<: *drep
+        Voting can be done together with delegation action or as a standalone action.
+      type: string
```

```suggestion
    ApiDRep: &ApiDRep
      description: |
        Decentralized representative (DRep)
        that the wallet is delegating its vote to.
        One can abstain, give no confidence vote,
        or vote for a representative specified by a key hash or script hash.
        Vote delegation can be done together with stake delegation action.
      nullable: false
      type: string
      oneOf:
        - enum:
            - abstain
            - no_confidence
          title: casting a default vote
        - <<: *drepKeyHash
            title: vote to a drep represented by key hash
        - <<: *drepScriptHash
            title: vote to a drep represented by script hash
```

Inlined, and amended the formulation slightly.

---

yes, sir